src/main/java/com/parkingapp/notificationservice/application/sendemailnotification/SendEmailNotificationUseCase.java

```diff
@@ -20,14 +21,16 @@ public SendEmailNotificationUseCase(
     }
 
     public void execute(UUID userId, UUID templateId) {
-        EmailTemplate emailTemplate = emailTemplateRepository.getEmailTemplateById(templateId);
+        Optional<EmailTemplate> emailTemplate = emailTemplateRepository.getEmailTemplateById(templateId);
         String userEmailAddress = "test@email.com";
-        EmailNotification emailNotification = new EmailNotification(
-                userEmailAddress,
-                emailTemplate.getSubject(),
-                emailTemplate.getBody()
-        );
-        this.emailService.send(emailNotification);
-    }
 
-}
+        if (emailTemplate.isPresent()) {
```

Comment:
i'm missing a test for the case it's not present

Reply:
done